modules/apps/static/portal-k8s-agent/portal-k8s-agent-test/src/testIntegration/java/com/liferay/portal/k8s/agent/internal/osgi/commands/test/ClientExtensionsOSGiCommandsTest.java

```diff
@@ -0,0 +1,407 @@
+		new LiferayIntegrationTestRule();
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		Bundle bundle = FrameworkUtil.getBundle(
+			ClientExtensionsOSGiCommandsTest.class);
+
+		_bundleContext = bundle.getBundleContext();
+
+		Company company = CompanyTestUtil.addCompany();
+
+		_companyWebId = company.getWebId();
+
+		_configurationPids = new ArrayList<>();
+
+		_configurationPids.add(
+			ConfigurationTestUtil.createFactoryConfiguration(
+				CETConfiguration.class.getName(),
+				"liferay-sample-cx-1/liferay.com",
+				HashMapDictionaryBuilder.<String, Object>put(
+					".cx.config.key",
+					CETConfiguration.class.getName() +
+						"~liferay-sample-cx-1/liferay.com"
+				).put(
+					"baseURL", "${portalURL}/o/liferay-sample-cx-1"
+				).put(
+					"dxp.lxc.liferay.com.virtualInstanceId", "default"
+				).put(
+					"name", "Liferay Sample CX 1"
+				).put(
+					"projectName", "liferay-sample-cx-1"
+				).put(
+					"test.only", "true"
+				).put(
+					"type", "customElement"
+				).build()));
+		_configurationPids.add(
+			ConfigurationTestUtil.createFactoryConfiguration(
+				CETConfiguration.class.getName(),
+				"liferay-sample-cx-2/liferay.com",
+				HashMapDictionaryBuilder.<String, Object>put(
+					".k8s.config.key",
+					CETConfiguration.class.getName() + "~liferay-sample-cx-2"
+				).put(
+					"baseURL", "${portalURL}/o/liferay-sample-cx-2"
+				).put(
+					"dxp.lxc.liferay.com.virtualInstanceId", "default"
+				).put(
+					"name", "Liferay Sample CX 2"
+				).put(
+					"projectName", "liferay-sample-cx-2"
+				).put(
+					"test.only", "true"
+				).put(
+					"type", "customElement"
+				).build()));
+		_configurationPids.add(
+			ConfigurationTestUtil.createFactoryConfiguration(
+				AccountEntryEmailConfiguration.class.getName(),
+				"liferay-sample-cx-3/" + _companyWebId,
+				HashMapDictionaryBuilder.<String, Object>put(
+					".k8s.config.key",
+					AccountEntryEmailConfiguration.class.getName() +
+						"~liferay-sample-cx-3/" + _companyWebId
+				).put(
+					"baseURL", "${portalURL}/o/liferay-sample-cx-3"
+				).put(
+					"dxp.lxc.liferay.com.virtualInstanceId", _companyWebId
+				).put(
+					"name", "Liferay Sample CX 3"
+				).put(
+					"projectName", "liferay-sample-cx-3"
+				).put(
+					"test.only", "true"
+				).put(
+					"type", "instanceSettings"
+				).build()));
+	}
+
+	@AfterClass
+	public static void tearDownClass() throws Exception {
+		for (String configurationPid : _configurationPids) {
+			ConfigurationTestUtil.deleteConfiguration(configurationPid);
+		}
+	}
+
+	@Test
+	public void testGetConfiguration() {
+		String pid =
+			CETConfiguration.class.getName() +
+				"~liferay-sample-cx-1/liferay.com";
+
+		Configuration configuration = _getConfiguration(pid);
+
+		Assert.assertEquals(pid, configuration.getPid());
+	}
+
+	@Test
+	public void testGetConfigurationNonexistentPid() {
+		String pid = "non-existent-pid";
```

Comment:
Inline `pid`, we are not re-using it anywhere, and it doesn't really increase readability much.